src/sdx/pce/topology/temanager.py

```diff
@@ -46,15 +51,38 @@ def add_topology(self, topology_data: dict):
         """
         self.topology_manager.add_topology(topology_data)
 
+    def update_topology(self, topology_data: dict):
+        """
+        Update an existing topology in TEManager.
+
+        :param topology_data: a dictionary that represents a topology.
+        """
+        self.topology_manager.update_topology(topology_data)
+
     def generate_connection_te(self) -> TrafficMatrix:
         """
         Generate a Traffic Matrix from the connection request we have.
         """
         ingress_port = self.connection.ingress_port
-        ingress_node = self.topology_manager.topology.get_node_by_port(ingress_port.id)
         egress_port = self.connection.egress_port
+
+        print(
+            f"generate_connection_te(), ports: "
+            f"ingress_port.id: {ingress_port.id}, "
+            f"egress_port.id: {egress_port.id}"
+        )
+
+        ingress_node = self.topology_manager.topology.get_node_by_port(ingress_port.id)
         egress_node = self.topology_manager.topology.get_node_by_port(egress_port.id)
 
+        if ingress_node is None:
+            print(f"No ingress node was found for ingress port ID '{ingress_port.id}'")
```

Comment:
Maybe these can go to logger, if we have the logger configured, otherwise we should keep them here for now, and add logger later.

Reply:
Both pce and datamodel do not do logging yet.  We have open issues to track them (https://github.com/atlanticwave-sdx/pce/issues/59, https://github.com/atlanticwave-sdx/datamodel/issues/10), and can address them after the demo.